### PR TITLE
Better pipeline

### PR DESCRIPTION
**Pull request by chrmod**
First version of complete jenkins pipeline for browser-f release

this PR must be squashed

### CODE REVIEW

**Comment by luciancor** [comment]
@alver-cliqz  squash merge please

File: win.Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  config.vm.guest = :windows
+  config.vm.boot_timeout = 500
+  config.vm.communicator = "winrm"
+  config.winrm.username = "cliqzfoxer"
+  config.winrm.password = "cliqzfoxer-245"
+  config.winrm.timeout = 21600 # 6 hours
+
+  config.vm.provider "vmware_workstation" do |v|
+    v.gui = false
+    v.memory = 4048
```
Comment (alver-cliqz) [inline win.Vagrantfile]: Not enough for PGO build

File: win.Vagrantfile [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    v.cpus = 6
+    v.vmx["RemoteDisplay.vnc.enabled"] = "true"
+    v.vmx["RemoteDisplay.vnc.port"] = "7905"
+  end
+
+  config.vm.provision "shell", run: "always", inline: <<-SHELL
+    cd c:/jenkins
+    Remove-Item slave.jar -ErrorAction SilentlyContinue
+    wget #{ENV['JENKINS_URL']}/jnlpJars/slave.jar -o slave.jar
```
Comment (alver-cliqz) [inline win.Vagrantfile]: What will be if resource not available?

Comment (chrmod) [inline win.Vagrantfile]: this one is served by jenkins, if it wont be there build probably had exploded already

File: win.Vagrantfile (hunk 1 above)
Comment (chrmod) [inline win.Vagrantfile]: was planning parametrization for this one. 

my question is, should we handle release builds with this PR or create a separate PR just after this one is merged?

Comment (alver-cliqz) [inline win.Vagrantfile]: Another PR will be ok

File: win.Vagrantfile (hunk 2 above)
Comment (alver-cliqz) [inline win.Vagrantfile]: Will see how this will work. If we will see a lot of issue here - will fix later

**Pull request merged**